actions/approve_csr_handler.go

```diff
@@ -100,17 +95,19 @@ func (h *approveCSRHandler) handle(ctx context.Context, log logrus.FieldLogger,
 	return errors.New("certificate signing request was not approved")
 }
 
-func (h *approveCSRHandler) getInitialNodeCSR(ctx context.Context, nodeName string) (*csr.Certificate, error) {
-	csrFetchCtx, cancel := context.WithTimeout(ctx, h.initialCSRFetchTimeout)
+func (h *approveCSRHandler) getInitialNodeCSR(ctx context.Context, log *logrus.Entry, nodeName string) (*csr.Certificate, error) {
+	log.Debug("getting initial csr")
+
+	ctx, cancel := context.WithTimeout(ctx, h.initialCSRFetchTimeout)
```

Comment:
question, we are basically replacing the original ctx timeout with this timeout, won't that cascade upstream? Shouldn't we have a single time out for this whole execution or maybe create a new context and not cancel the main context because of this failure?

Reply:
we are not replacing it, it will cascade downstream with this timeout